va/drm/va_drm_utils.c

```diff
@@ -110,6 +112,14 @@ VA_DRM_GetDriverName(VADriverContextP ctx, char **driver_name_ptr, int candidate
         return VA_STATUS_ERROR_INVALID_DISPLAY;
 
     *driver_name_ptr = va_DRM_GetDrmDriverName(drm_state->fd);
+
+    /* Map vgem to WSL2 for Windows subsystem for linux */
+    struct utsname sysinfo = {};
+    if (!strncmp(*driver_name_ptr, "vgem", 4)
+        && !uname(&sysinfo)
```

Comment:
from `man 3 uname`:
> Upon successful completion, a non-negative value shall be returned.

I.e. it might necessarily return 0 on success depending on implementation.

Also, does VS support this function?

Reply:
Gotcha, I was reading from [here](https://man7.org/linux/man-pages/man2/uname.2.html) instead, to make sure I added `&& (uname(&sysinfo) >= 0)` which should work with both definitions.

This will not build for windows, it's using libva in [Windows Subsystem for linux](https://learn.microsoft.com/en-us/windows/wsl/about#what-is-wsl-2). The goal of this change is to automatically detect and load the d3d12_drv_video.so mesa gallium driver instead of having to always set `LIBVA_DRIVER_NAME=d3d12`

Reply:
More context: We enabled libva in WSL/Windows native last year by adding an d3d12-based implementation of the mesa VA driver. When using that driver on Windows, it is named `vaon12_drv_video.dll` and uses the libva-win32 backend. In WSL it's named `d3d12_drv_video.so` and uses the usual backends like DRM.